src/coreclr/jit/codegencommon.cpp

```diff
@@ -864,11 +864,11 @@ BasicBlock* CodeGen::genCreateTempLabel()
 #endif
 
     JITDUMP("Mark " FMT_BB " as label: codegen temp block\n", block->bbNum);
-    block->bbFlags |= BBF_HAS_LABEL;
+    block->SetFlag(BBF_HAS_LABEL);
 
     // Use coldness of current block, as this label will
     // be contained in it.
-    block->bbFlags |= (compiler->compCurBB->bbFlags & BBF_COLD);
+    block->SetFlag(compiler->compCurBB->bbFlags & BBF_COLD);
```

Comment:
You could add a `CopyFlags` function and have:
```suggestion
    block->CopyFlags(compiler->compCurBB, BBF_COLD);
```

Reply:
Sure thing. We do have a few cases where the second argument to `CopyFlags` would be multiple flags OR'd together, e.g. in `Compiler::fgSplitBlockBeforeTree`:
`block->SetFlag(originalFlags & (BBF_SPLIT_GAINED | BBF_IMPORTED | BBF_GC_SAFE_POINT | BBF_LOOP_PREHEADER | BBF_RETLESS_CALL));`
Would you be ok with this pattern?

Reply:
Something like:
```
block->CopyFlags(fromBlock, BBF_IMPORTED | BBF_GC_SAFE_POINT | BBF_LOOP_PREHEADER | BBF_RETLESS_CALL);
```
would be fine.

The more complex cases where we (1) grab the flags, (2) do some computations, then (3) set the flags, might be where the `GetFlagsRaw` would come in. So the example you reference might need to use that.